### PR TITLE
CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by vsalvino**
Ok, I went a little crazy playing with Azure Pipelines.

This change adds CI via Azure Pipelines, which is free for open source projects. I created an Azure DevOps project called "CodeRed CMS" within our organization, and connected it to this GitHub repo.

The `azure-pipelines.yml` file defines the CI instructions, very similar to how Travis works.

One really cool thing with Azure is that they have viewers for unit test reports and coverage reports. So this PR also changes the way test are run. See the contributing guide diff in this PR for both ways to run tests.

You can also see a stripped down preview in the **Checks** tab on this PR. The full ultra-detailed output is also publicly available from the link below. Click the **Tests** and **Coverage** tabs to play with the reporting interface.
https://dev.azure.com/coderedcorp/coderedcms/_build/results?buildId=23

I mostly made this just to learn Azure Pipelines, but it actually might be worth switching from Travis + Codecov. Thoughts?

### CODE REVIEW

**Comment by vsalvino** [comment]
Ok, lots more playing and optimization. I'm satisfied with the pipeline now. Main changes:
* only flakes the diff between your PR and master (or flake everything if you're on the master branch)
* If the project template changed, then flake the testproject as well.
* moved this logic into a powershell script which can be run locally on windows/mac/linux as well.

File: docs/contributing/index.rst [hunk 1]
```diff
@@ -92,11 +92,19 @@ license header comment states copyright, ownership, license, and also provides c
 Testing CodeRed CMS
 -------------------
 
-To run the built in tests for CodeRed CMS, run the following in your test project's directory:
+To run the built in tests for CodeRed CMS using Django, run the following:
 
 .. code-block:: console
 
-    $ python manage.py test coderedcms --settings=coderedcms.tests.settings
+    $ python testproject/manage.py test coderedcms --settings=coderedcms.tests.settings
```
Comment (corysutyak) [inline docs/contributing/index.rst]: Is there a particular reason for this change?

**Comment by corysutyak** [comment]
I'm impartial to which CI system we use as I don't have the DevOps experience to differentiate between the two, so this seems fine to me.

**Comment by WMeldon** [comment]
Looks good.  Might consider adding the badge to the README.

```
[![Build Status](https://dev.azure.com/coderedcorp/coderedcms/_apis/build/status/coderedcorp.coderedcms?branchName=master)](https://dev.azure.com/coderedcorp/coderedcms/_build/latest?definitionId=1&branchName=master)
```

**Comment by vsalvino** [comment]
> Looks good.  Might consider adding the badge to the README.
> [Build Status](https://dev.azure.com/coderedcorp/coderedcms/_build/latest?definitionId=1&branchName=master)

Good call... although master is going to fail hard now that this is merged. We can add the badge after Kevin does cleanup.

**Pull request merged**